components/ui/pathogen-map/pathogen-country-highlight-layer.tsx

```diff
@@ -0,0 +1,102 @@
+import { Layer, Source } from "react-map-gl";
+import { PathogenDataPointPropertiesBase } from "./pathogen-map";
+import { PathogenMapLayerInfoWithCountryHighlighting } from "./pathogen-map-layer";
+import { useEffect, useState } from "react";
+import { getEsriVectorSourceStyle } from "@/utils/mapping-util";
+import {
+  MapResources,
+  MapSymbology,
+} from "@/app/pathogen/arbovirus/dashboard/(map)/map-config";
+import { countryNameToIso31661Alpha3CodeMap } from "@/lib/country-iso-3166-1-alpha-3-codes";
+
+interface PathogenCountryHighlightLayerProps<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> {
+  dataLayer:
+    | PathogenMapLayerInfoWithCountryHighlighting<TPathogenDataPointProperties>
+    | undefined;
+  positionedUnderLayerWithId: string | undefined;
+}
+
+const generatePaintForLayer = <
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+>(input: {
+  dataLayer:
+    | PathogenMapLayerInfoWithCountryHighlighting<TPathogenDataPointProperties>
+    | undefined;
+}) => {
+  const { dataLayer } = input;
+
+  const allUniqueCountryCodesWithData = new Set(
+    dataLayer?.dataPoints
+      .map((dataPoint) => {
+        return countryNameToIso31661Alpha3CodeMap[dataPoint.country];
+      })
+      .filter(
+        (alpha3CountryCode: string | undefined): alpha3CountryCode is string =>
+          !!alpha3CountryCode
+      ) ?? []
+  );
+
+  const countryColorsAndOpacities = Array.from(
+    allUniqueCountryCodesWithData
+  ).map((alpha3CountryCode) => ({
+    alpha3CountryCode: alpha3CountryCode,
+    fill: MapSymbology.CountryFeature.HasData.Color,
+    opacity: MapSymbology.CountryFeature.HasData.Opacity,
```

Comment:
Would be down to turn the opacity down a little bit or make the colour a little bit less harsh, right now it's 0.5 and you can see what that looks like in the gif (maybe a bit harsh with the pastel colours?)